src/redata.js

```diff
@@ -1,15 +1,31 @@
+/**
+ * redata's data object definition
+ * @typedef {Object} Data
+ * @property {boolean} loading - true if the loader is still running, false otherwise.
+ * @property {error} error - instance of Error in case the loader failed, undefined otherwise.
+ * @property {*} result - result of the loader, or undefined if the loader is still running.
+ */
+
+/**
+ * @param {function} loader - called to fetch new data
+ * @param {function} [shouldReload = defaultShouldReload] - decides whether a redata should occur
+ * @param {function} [mapper = defaultMapper] - maps the {@link Data} object values to your component's props
+ * @param {object} [initialCtx = defaultInitialCtx] - starting point of redata, defines if there's any preloaded data
+ * @returns {function} redata wrapper function
```

Comment:
Regarding the new line, I'm not enforcing it. I'm used to it since other projects I have seen do use new line to separate block types. @marcooliveira ?